### DISCUSSION TITLE
Segmentation fault(core dumped)

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#25797] [Q&A General]
I am trying to run step1.i input file. Below is the output I get. Is there a way to solve this?
```
[emmanuel324@tinkercliffs1 problems]$ ../darcy_thermo_mech-opt -i step1.i 
Segmentation fault (core dumped)
```

**Comment by GiudGiud** [comment]
Hello

Did you create your own object for that input file?
If you want to debug it you can follow the instructions:
https://mooseframework.inl.gov/application_development/debugging.html
If you built your own objects please paste the source and header here


Guillaume

**Reply by Emmanuel-324** [reply]
No I was just trying to run the step1.i input file from the moose website in my directory.

**Reply by GiudGiud** [reply]
ok then clean the repo with `git clean -xfd` in the root (moose/), recompile the executable and try again.

**Reply by Emmanuel-324** [reply]
Do you mean I should run it in the moose directory?

**Reply by GiudGiud** [reply]
yes the clean command only cleans folder below it so it needs to be run in moose/

**Reply by Emmanuel-324** [reply]
I did run it in my moose directory. And when you said I should compile an executable do you mean this line:
cd ~/projects/belson
make -j 6
./run_tests -j 6


or this
.../darcy_thermo_mech-opt -i step.i

**Reply by GiudGiud** [reply]
the first one compiles then runs tests

**Reply by Emmanuel-324** [reply]
Is it this
cd ~emmanuel324/projects/moose/tutorials/darcy_thermo_mech
make make -j4

**Reply by GiudGiud** [reply]
`make -j4`
but yes

**Reply by Emmanuel-324** [reply]
When I try make an executable in the directory that has the makefile this is the feedback I get.

[emmanuel324@tinkercliffs1 step01_diffusion]$ make -j4
make: mpicxx: Command not found
make: mpif90: Command not found
/bin/sh: mpif90: command not found
Rebuilding symlinks in /home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/step01_diffusion/build/header_symlinks
Linking Library /home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/step01_diffusion/test/lib/libdarcy_thermo_mech_test-opt.la...
Traceback (most recent call last):
  File "/home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/../..//framework/scripts/write_appresource_file.py", line 6, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
Creating Resource file /home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/step01_diffusion/darcy_thermo_mech.yaml
Compiling C++ (in opt mode) /home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/step01_diffusion/src/base/DarcyThermoMechApp.C...
Compiling C++ (in opt mode) /home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/step01_diffusion/src/main.C...
/home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/../..//libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/step01_diffusion/src/main.x86_64-pc-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....
/home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/../..//libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/step01_diffusion/src/base/DarcyThermoMechApp.x86_64-pc-linux-gnu.opt.lo] Error 1
/home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/../..//libmesh/installed/contrib/bin/libtool: line 10545: mpicxx: command not found
make: *** [/home/emmanuel324/projects/moose/tutorials/darcy_thermo_mech/step01_diffusion/test/lib/libdarcy_thermo_mech_test-opt.la] Error 127

**Reply by GiudGiud** [reply]
You forgot to activate the moose conda environment

**Reply by Emmanuel-324** [reply]
How do I activate the conda environment before I create the executable.  Ps: I am using the MOOSE on an HPC cluster

**Reply by GiudGiud** [reply]
So you did not use conda / mamba to install MOOSE?

**Reply by Emmanuel-324** [reply]
No I did the installation through an HPC cluster.

**Reply by GiudGiud** [reply]
If you did, 

mamba activate moose 

If you did not you need to load the modules you used to compile the first time

**Reply by Emmanuel-324** [reply]
Thanks. It worked!
